cytotable/utils.py

```diff
@@ -600,3 +570,54 @@ def evaluate_futures(sources: Union[Dict[str, List[Dict[str, Any]]], str]) -> An
         if isinstance(sources, dict)
         else _unwrap_value(sources)
     )
+
+
+def _generate_pagesets(
+    keys: List[Union[int, float]], chunk_size: int
+) -> List[Tuple[Union[int, float], Union[int, float]]]:
+    """
+    Generate a pageset (keyset pagination) from a list of keys.
+
+    Parameters:
+        keys List[Union[int, float]]:
+            List of keys to paginate.
+        chunk_size int:
+            Size of each chunk/page.
+
+    Returns:
+        List[Tuple[Union[int, float], Union[int, float]]]:
+            List of (start_key, end_key) tuples representing each page.
+    """
+
+    # if we have no keys, raise an exception
+    if not keys:
+        raise CytoTableException("No keys were provided to pageset generation process.")
```

Comment:
Can you switch to active voice? Would a user provide the keys? Would some function generate them?

Reply:
Thanks, yes, I'll switch this to active voice. At the moment keys must be provided by a user via the `page_keys` parameter and eventually reach this function to help create "pagesets" which may be processed independently of one another.

Reply:
Looking into this more I realized it might not be possible to reach this exception case given the design of the function and how `convert` works. I ended up removing it instead to keep things simple.